code/modules/cm_marines/equipment/kit_boxes.dm

```diff
@@ -257,6 +276,31 @@
 		return TRUE
 	return FALSE
 
+//Grenader weapon selector snowflake
+/obj/item/spec_kit/grenader_weapon_selector
+	name = "\improper Grenader weapon case"
+	desc = "A large case with the option of the M92 grenade launcher and the P9 SHARP Rifle."
+
+/obj/item/spec_kit/grenader_weapon_selector/can_use(mob/living/carbon/human/user)
+	return TRUE
+
+/obj/item/spec_kit/grenader_weapon_selector/select_and_spawn(mob/living/carbon/human/user)
+	var/grenader_selection = tgui_input_list(usr, "Select your weapon", "Weapon Selection", list("M92 Grenade Launcher", "P9 SHARP Rifle"))
+	if(!grenader_selection || QDELETED(src))
+		return FALSE
+	var/turf/T = get_turf(loc)
+	var/obj/item/storage/box/spec/weapon_box
```

Comment:
why do you need to define this as a var when you don't use it (those weapon box = new x below can just be new x)

Reply:
*glances at the code 5 lines above it!!!*

Reply:
need to put in hands (very important!!!)